packages/internal-playwright-helpers/src/fixtures.ts

```diff
@@ -0,0 +1,60 @@
+import { Page, test as base } from "@playwright/test";
+import { getBrowserTestingEnvironment } from "@inrupt/internal-test-env";
+import { TestPage } from "./pages/testApp";
+import { OpenIdPage } from "./pages/open-id";
+import { CognitoPage } from "./pages/cognito";
+
+export type TestOptions = {
+  loginOnStart: boolean;
+};
+
+/**
+ * Start the app, and if applicable initiate login from the client, get redirected
+ * to the OpenID provider, log in, allow the client when redirected to the broker,
+ * and completes login after being redirected to client.
+ */
+const initializeTestApp = async (
+  page: Page,
+  openidProvider: string,
+  loginOnStart: boolean,
+  login: string,
+  password: string
+): Promise<void> => {
+  const testPage = new TestPage(page, openidProvider);
+  // Note: these steps must execute in series, not parallel, which is what
+  // Promise.all would do:
+  await testPage.start()
+  if (loginOnStart) {
+    const cognitoPage = new CognitoPage(page);
+    const openIdPage = new OpenIdPage(page);
+    await testPage.startLogin();
+    await cognitoPage.login(login, password);
+    await openIdPage.allow();
+    await testPage.handleRedirect();
+  }
+};
```

Comment:
A better split might be to do:

```
const testPage = new TestPage(page, openidProvider);
await testPage.start()
```

In the `page` method of the Fixture, with just the login on start being whether you call `loginAndAllow` on TestPage or not.

Reply:
So the lines 28-33 would be inside a method on `TestPage` (loginAndAllow), and there's essentially no need for this method

Reply:
That's how I initially approached, but it felt cleaner to have the test page be a page model similar to the others, without any extra dependencies, and then to have the multi-page coordination done at the fixture level. This would mean that adding new providers such as KeyCloak would require no change in the test page, and only in the fixtures, which seems like the right place.